x/multi-staking/keeper/proposal.go

```diff
@@ -0,0 +1,52 @@
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/realio-tech/multi-staking-module/x/multi-staking/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// AddMultiStakingCoinProposal handles the proposals to add a new bond token
+func (k Keeper) AddMultiStakingCoinProposal(
+	ctx sdk.Context,
+	p *types.AddMultiStakingCoinProposal,
+) error {
+	_, found := k.GetBondWeight(ctx, p.Denom)
+	if found {
+		return fmt.Errorf("Error MultiStakingCoin %s already exist", p.Denom) //nolint:stylecheck
+	}
+
+	k.SetBondWeight(ctx, p.Denom, *p.BondWeight)
```

Comment:
We need to add a validation for checking `BondWeight` value is greater than zero.